atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java

```diff
@@ -104,6 +105,15 @@ protected void sanityCheckTableName(TableReference tableRef) {
                 "invalid tableName: " + tableName);
     }
 
+    protected void sanityCheckTableMetadata(TableReference tableRef, byte[] tableMetadata) {
+        if (tableMetadata == null || Arrays.equals(tableMetadata, AtlasDbConstants.EMPTY_TABLE_METADATA)) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Passing in empty table metadata for table '%s' does not make sense, and is disallowed.",
```

Comment:
This message could be less snarky :)
Could also recommend using `AtlasDbConstants.GENERIC_TABLE_METADATA`?